### PR TITLE
Add support for rmw_connextdds

### PR DESCRIPTION
**Pull request by asorbini**
This PR replaces all references to `rmw_connext_cpp` (and `rmw_connext_dynamic_cpp`) with `rmw_connextdds`.

See [rticommunity/rmw_connextdds #9](https://github.com/rticommunity/rmw_connextdds/issues/9) for a list of related PRs, and an overview of all the changes required to replace [ros2/rmw_connext](https://github.com/ros2/rmw_connext) (`rmw_connext_cpp`) with [rticommunity/rmw_connextdds](https://github.com/rticommunity/rmw_connextdds) in the ROS2 source tree.

### CODE REVIEW

**Review by fujitatomoya — APPROVED** [review]

File: test_security/CMakeLists.txt [hunk 1]
```diff
@@ -256,7 +256,7 @@ if(BUILD_TESTING)
     set(ENV_PATH "$ENV{PATH}")
     file(TO_CMAKE_PATH "${ENV_PATH}" ENV_PATH)
     set(TEST_PATH "${ENV_PATH}")
-    if(rmw_implementation STREQUAL "rmw_connext_cpp")
+    if(rmw_implementation STREQUAL "rmw_connextdds")
```
Comment (ivanpauno) [inline test_security/CMakeLists.txt:259]:
```suggestion
    if(rmw_implementation MATCHES "rmw_connext(.*)")
```

Comment (asorbini) [reply test_security/CMakeLists.txt]: Applied in [7069da0](https://github.com/ros2/system_tests/pull/463/commits/7069da00fed8ae1fd00ba97d108943df82f7319d).

**Review by ivanpauno — APPROVED** [review]

**Comment by ivanpauno** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13899/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8750/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11608/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13990/)

File: test_security/CMakeLists.txt (hunk 1 above)
Comment (ivanpauno) [reply test_security/CMakeLists.txt]: mmm didn't realize before but https://github.com/ros2/system_tests/pull/463/commits/7069da00fed8ae1fd00ba97d108943df82f7319d doesn't look correct, `STREQUAL` doesn't work like `MATCHES` so the `if()` is never satisfied.

No idea way, but CI is still happy with this, it seems that this `if()` is not more needed (?).

Comment (asorbini) [reply test_security/CMakeLists.txt]: Ha! My bad for being too hasty in making the "correction". Thank you for catching it!

Re: CI, maybe the directory is already added somewhere else?

Comment (ivanpauno) [reply test_security/CMakeLists.txt]: > Re: CI, maybe the directory is already added somewhere else?

maybe, can it be this https://github.com/rticommunity/rmw_connextdds/blob/b88131f28f7e6e6b0f5dff6eb9be133e512eccfd/rti_connext_dds_cmake_module/env_hook/rti_connext_dds.sh.in#L131? (or sth else that file is doing)

Anyways, it seems safer to keep the condition for the moment.

**Comment by ivanpauno** [comment]
another run of CI, just in case:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13911/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8760/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11618/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14000/)

File: test_security/CMakeLists.txt (hunk 1 above)
Comment (asorbini) [reply test_security/CMakeLists.txt]: > maybe, can it be this https://github.com/rticommunity/rmw_connextdds/blob/b88131f28f7e6e6b0f5dff6eb9be133e512eccfd/rti_connext_dds_cmake_module/env_hook/rti_connext_dds.sh.in#L131? (or sth else that file is doing)
> 

It's possible, although I would have expected `RTI_OPENSSL_BIN` to point to something other than `$NDDSHOME/bin`. It ultimately comes down to what the value of that variable is.

> Anyways, it seems safer to keep the condition for the moment.

Agreed.

**Comment by ivanpauno** [comment]
Thanks @asorbini !

**Pull request merged**